includes/admin/OrderSettings.php

```diff
@@ -246,6 +259,70 @@ private function setAllData(): void
         $this->setDigitalStampRangeWeight();
     }
 
+    // TODO refactor this (MY-28781)
+    public function setShippingRecipient(): self
+    {
+        $thisOrder  = $this->order;
+        $streetName = $this->stringConcatForAddressLine(
+            40,
+            $thisOrder->get_shipping_address_1(),
+            $thisOrder->get_shipping_address_2()
+        );
+
+        $this->shippingRecipient = (new Recipient())
+            ->setCc($thisOrder->get_shipping_country())
+            ->setCity($thisOrder->get_shipping_city())
+            ->setCompany($thisOrder->get_shipping_company())
+            ->setPerson(substr($thisOrder->get_formatted_shipping_full_name(), 0, 50))
+            ->setPostalCode($thisOrder->get_shipping_postcode())
+            ->setStreet($streetName);
+
+        return $this;
+    }
+
+    public function getShippingRecipient(): ?Recipient
+    {
+        return $this->shippingRecipient;
+    }
+
+    public function setBillingRecipient(): self
+    {
+        $thisOrder  = $this->order;
+        $streetName = $this->stringConcatForAddressLine(
+            40,
+            $thisOrder->get_billing_address_1(),
+            $thisOrder->get_billing_address_2()
+        );
+
+        $this->billingRecipient = (new Recipient())
+            ->setCc($thisOrder->get_billing_country())
+            ->setCity($thisOrder->get_billing_city())
+            ->setCompany($thisOrder->get_billing_company())
+            ->setEmail($thisOrder->get_billing_email())
+            ->setPerson(substr($thisOrder->get_formatted_billing_full_name(), 0, 50))
+            ->setPhone($thisOrder->get_billing_phone())
+            ->setPostalCode($thisOrder->get_billing_postcode())
+            ->setStreet($streetName);
+
+        return $this;
+    }
+
+    public function getBillingRecipient(): ?Recipient
+    {
+        return $this->shippingRecipient;
+    }
+
+    /**
+     * @param int    $maxLengthReturned
+     * @param string ...$parts
+     *
+     * @return string
+     */
+    private function stringConcatForAddressLine(int $maxLengthReturned = 255, string ...$parts): string
+    {
+        return substr(trim(implode(' ', $parts)), 0, $maxLengthReturned) ?: '';
```

Comment:
Wat doet dit? De naam is niet echt een werkwoord of iets gerelateerds. Je noemt hem ook maar 2x aan en allebei met dezelfde 1e parameter.